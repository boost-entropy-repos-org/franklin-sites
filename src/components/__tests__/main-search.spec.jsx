import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import MainSearch from '../main-search';

Enzyme.configure({ adapter: new Adapter() });

describe('MainSearch component', () => {
  const handleSubmit = jest.fn();
  const handleChange = jest.fn();

  test('should submit the search term', () => {
    const wrapper = mount(<MainSearch onChange={handleChange} onSubmit={handleSubmit} />);
    wrapper.find('[type="submit"]').simulate('submit');
    expect(handleSubmit).toHaveBeenCalled();
  });

  test('should detect the search term', () => {
    const wrapper = mount(<MainSearch onChange={handleChange} onSubmit={handleSubmit} />);
    wrapper
      .find('[type="text"]')
      .first()
      .simulate('change', { target: { value: 'foo' } });
    expect(handleChange).toHaveBeenCalledWith('foo');
  });

  test('should set searchTerm', () => {
    const wrapper = shallow(
      <MainSearch searchTerm="blah" onChange={handleChange} onSubmit={handleSubmit} />,
    );
    expect(
      wrapper
        .find('[type="text"]')
        .first()
        .props().value,
    ).toBe('blah');
  });
});
