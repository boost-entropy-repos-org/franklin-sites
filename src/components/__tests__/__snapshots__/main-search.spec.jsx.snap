// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSearch component should render 1`] = `
<DocumentFragment>
  <form
    class="main-search"
    data-testid="main-search-form"
    style="--main-button-color: #00639a;"
  >
    <input
      class="main-search__input"
      data-testid="main-search-input"
      type="text"
      value=""
    />
    <button
      class="button primary"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;

exports[`MainSearch component should render with namespaces selector 1`] = `
<DocumentFragment>
  <form
    class="main-search"
    data-testid="main-search-form"
    style="--main-button-color: #00639a;"
  >
    <div
      class="dropdown-container"
    >
      <button
        class="button primary dropdown"
        type="button"
      >
        One
      </button>
      <div
        class="dropdown-menu"
      />
    </div>
    <input
      class="main-search__input"
      data-testid="main-search-input"
      type="text"
      value=""
    />
    <button
      class="button primary"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
